service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -110,11 +110,21 @@ private RunSetDetailsResponse convertToRunSetDetails(RunSet runSet) {
   public ResponseEntity<RunSetListResponse> getRunSets(UUID methodId, Integer pageSize) {
     List<RunSet> runSets = runSetDao.getRunSets();
     List<RunSet> updatedRunSets = smartRunSetsPoller.updateRunSets(runSets);
-
-    List<RunSetDetailsResponse> runSetDetails =
-        updatedRunSets.stream().map(this::convertToRunSetDetails).toList();
-    RunSetListResponse response = new RunSetListResponse().runSets(runSetDetails);
-
+    RunSetListResponse response;
+
+    if (methodId != null) {
+      List<RunSet> filteredRunSet =
+          updatedRunSets.stream()
```

Comment:
Instead of filtering in memory, I would be tempted to replace `runSetDao.getRunSets();` with (a new) `runSetDao.getRunSetsForMethod(methodId, pageSize);` call.

The reason is, if someone searches for run sets for a particular method, it feels a little weird to me that we would run the smart poller for _all_ run sets, then filter down in memory, and then return the result to them.

Reply:
This would also let us respect the `pageSize` option.